nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -953,14 +962,21 @@ ngx_http_modsecurity_init(ngx_conf_t *cf)
     }
     *h = ngx_http_modsecurity_handler;
 
-    ngx_http_next_header_filter = ngx_http_top_header_filter;
-    ngx_http_top_header_filter = ngx_http_modsecurity_header_filter;
-
-    ngx_http_next_body_filter = ngx_http_top_body_filter;
-    ngx_http_top_body_filter = ngx_http_modsecurity_body_filter;
-
     ngx_memzero(&ngx_http_modsecurity_upstream, sizeof(ngx_http_upstream_t));
     ngx_http_modsecurity_upstream.cacheable = 1;
+
+    ngx_thread_pool_t* thread_pool = ngx_thread_pool_add(cf, &thread_pool_name);
```

Comment:
How are we limiting the max thread count?

Reply:
Is it max_queue?

Reply:
There's a fixed number of workers who work on tasks that get enqueued. The default in nginx is 32. It can be configured in nginx.conf like this
thread_pool default threads=128 max_queue=65536;